packages/ptokens-assets-evm/src/ptokens-evm-asset-builder.ts

```diff
@@ -30,6 +30,25 @@ export class pTokensEvmAssetBuilder extends pTokensAssetBuilder {
       method: 'hub',
       abi: factoryAbi,
     })
+    const address =
+      this._assetInfo.underlyingAssetNetworkId === this._assetInfo.networkId &&
```

Comment:
This check should be isNative(), isn't it?
Anyway, in case of a ptoken, assetTokenAddress would be undefined.
As a consequence, the field in AssetInfo needs to be optional, and you may evaluate it in the builder, i.e. the field `address` is not needed, as its function is already covered by AssetInfo's `assetTokenAddress`.

Reply:
Yes, correct. 
The check should be isNative but it is not available yet. I changet it to just check if assetTokenAddress exists in the first place. If it does not it is a ptoken. The rest is updated.